examples/asg_service_linked_role/main.tf

```diff
@@ -0,0 +1,124 @@
+provider "aws" {
+  region = "eu-west-1"
+
+  # Make it faster by skipping something
+  skip_get_ec2_platforms      = true
+  skip_metadata_api_check     = true
+  skip_region_validation      = true
+  skip_credentials_validation = true
+  skip_requesting_account_id  = true
+}
+
+##############################################################
+# Data sources to get VPC, subnets and security group details
+##############################################################
+data "aws_vpc" "default" {
+  default = true
+}
+
+data "aws_subnet_ids" "all" {
+  vpc_id = data.aws_vpc.default.id
+}
+
+data "aws_security_group" "default" {
+  vpc_id = data.aws_vpc.default.id
+  name   = "default"
+}
+
+data "aws_ami" "amazon_linux" {
+  most_recent = true
+  owners      = ["137112412989"] # Amazon
+
+  filter {
+    name = "name"
+
+    values = [
+      "amzn-ami-hvm-*-x86_64-gp2",
+    ]
+  }
+
+  filter {
+    name = "owner-alias"
+
+    values = [
+      "amazon",
+    ]
+  }
+}
+
+resource "aws_iam_service_linked_role" "autoscaling_service_linked_role" {
+  aws_service_name = "autoscaling.amazonaws.com"
+  description = "A service linked role for autoscaling groups"
+  custom_suffix = "TEST"
+
+  lifecycle {
+    ignore_changes = ["custom_suffix"]
```

Comment:
Could you please explain why `cusomt_suffix` should be ignored here?

Reply:
Sure I'll add a comment within the PR tomorrow. Basically Terraform will recreate the resource on every run if you don't ignore the custom_suffix parameter. 

From: https://www.terraform.io/docs/providers/aws/r/iam_service_linked_role.html

> custom_suffix - (Optional, forces new resource) Additional string appended to the role name. Not all AWS services support custom suffixes.

Reply:
Is there anything else that is missing or you would like to see for this PR to be merged in?